### PR TITLE
Hcap 78 workforce baseline

### PR DESCRIPTION
**Pull request by mauriciofreshw**

### CODE REVIEW

File: client/src/components/employer-form/BaselineList.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  const convertValue = (value) => {
+    return typeof value === "undefined" ? '' : value
+  }
+
+  return (
+    <Fragment>
+      {options.map((option) => (
+        <Fragment key={option.value}>
+          <Accordion
+            expanded={value[option.value]?.add ? true : false}
```
Comment (fwkendall) [inline client/src/components/employer-form/BaselineList.js:50]: is this ternary necessary or can this just cast `value[option.value]?.add` to a boolean?

File: client/src/components/employer-form/BaselineList.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                ...value[option.value],
+                add: !value[option.value]?.add,
+              });
+            }}
+            {...props}
+          >
+            <AccordionSummary expandIcon={
+              !value[option.value]?.add ?
+                <AddCircleIcon color="primary" />
+                :
```
Comment (fwkendall) [inline client/src/components/employer-form/BaselineList.js:62]: interesting, is putting this on its own line a react convention? in other languages I've used it's usually inline so just for my future reference :)

File: client/src/components/employer-form/BaselineList.js (hunk 1 above)
Comment (mauriciofreshw) [reply client/src/components/employer-form/BaselineList.js:50]: `expanded` prop does not accept **undefined**

File: client/src/components/employer-form/index.js [hunk 3]
```diff
@@ -13,7 +14,8 @@ import { Summary } from './Summary';
 import { OperatorInfo } from './OperatorInfo';
 import { SiteInfo } from './SiteInfo';
 import { HcapRequest } from './HcapRequest';
-import { BusinessDetails } from './BusinessDetails';
+
```
Comment (fwkendall) [inline client/src/components/employer-form/index.js:17]: so minor, but I don't think we want this newline :) the other form components are all grouped above

File: client/src/components/employer-form/index.js [hunk 4]
```diff
@@ -52,19 +54,37 @@ export const Form = ({ initialValues, isDisabled }) => {
...
+    let newWorkforceBaseline = [];
+
+    mapValues(values.workforceBaseline, (value, key) => {
+      if (value.add) {
+        newWorkforceBaseline.push({
+          role: key,
+          currentFullTime: value.currentFullTime,
+          currentPartTime: value.currentPartTime,
+          currentCasual: value.currentCasual,
+          vacancieFullTime: value.vacancieFullTime,
```
Comment (fwkendall) [inline client/src/components/employer-form/index.js:71]: should this be _vacancies_ or _vacancy_ maybe? Looks like it was meant to be plural and there's a typo (the issue is present in BaselineList also, could probably do a global search/replace)

File: client/src/components/employer-form/BaselineList.js (hunk 2 above)
Comment (mauriciofreshw) [reply client/src/components/employer-form/BaselineList.js:62]: Depends on your line size, usually good lint doesn’t allow lines too big (100-200), so I'm used to not write lines too big :)

File: client/src/components/employer-form/index.js (hunk 4 above)
Comment (mauriciofreshw) [reply client/src/components/employer-form/index.js:71]: good catch!

File: client/src/constants/validation.js [hunk 5]
```diff
@@ -111,10 +107,26 @@ export const EmployerFormSchema = yup.object().noUnknown('Unknown field for form
...
-  // TODO - Business details from mock
-  businessKind: yup.string().required(errorMessage),
-  workersSize: yup.number().required(errorMessage).integer('Number of workers must be an integer').moreThan(0, 'Number must be greater than 0'),
-  employerType: yup.string().required(errorMessage),
+  // Workforce Baseline
+  workforceBaseline: yup.lazy(obj => yup.object()
+    .shape(
+      mapValues(obj, (value, key) => {
+        if (!value.add) {
+          return yup.object().noUnknown('Unknown field for work baseline').shape({
```
Comment (fwkendall) [inline client/src/constants/validation.js:115]: should this be workforce? same question for line 119

**Review by fwkendall — DISMISSED** [review]
Looks great, nice work on this 👍 just a few comments/questions from me, I think I spotted a couple of typos (although only one is user-facing) 😄

File: client/src/components/employer-form/BaselineList.js (hunk 1 above)
Comment (fwkendall) [reply client/src/components/employer-form/BaselineList.js:50]: gotcha, I was hoping it would be truthy but sounds like no

**Review by fwkendall — APPROVED** [review]

**Review by fwkendall — APPROVED** [review]

File: client/src/components/employer-form/BaselineList.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+  const setValue = (option, result) => {
+    const mergedResult = { ...value, [option.value]: result }
+    setFieldValue(name, mergedResult);
+  };
+
+  const classes = useStyles();
+  const error = form.errors[name];
+
+  const convertValue = (value) => {
```
Comment (ashtonmeuser) [inline client/src/components/employer-form/BaselineList.js:41]: Nitpick: `convertValue` is a little ambiguous. It's almost more of a `sanitizeValue`.

**Review by ashtonmeuser — APPROVED** [review]

**Pull request merged**